File: 482.密钥格式化.py
#
# @lc app=leetcode.cn id=482 lang=python
#
# [482] 密钥格式化
#

# @lc code=start
class Solution(object):
    def licenseKeyFormatting(self, S, K):
        """
        :type S: str
        :type K: int
        :rtype: str
        """
        s = S.upper().replace('-', '')[::-1]
        res = '' 
        for i in range(len(s)):
            if i % K == 0 and i != 0:
                res = '-' + res
            res = s[i] + res

        return res



                        
        
# @lc code=end

File: 482.密钥格式化.cpp
/*
 * @lc app=leetcode.cn id=482 lang=cpp
 *
 * [482] 密钥格式化
 */

// @lc code=start
class Solution {
public:
    string licenseKeyFormatting(string S, int K) {
        stack<char> stk;
        for(int i = 0; i < S.size(); i++){
            if(S[i] != '-')
                stk.push(toupper(S[i]));
            S[i] = toupper(S[i]);
        }
        string s = "";
        int count = 0;
        int n = stk.size();
        int need = n / K;
        while(!stk.empty())
        {
            count++;
            char tmp = stk.top();
            stk.pop();
            s += tmp;
            if(count % K == 0 and !stk.empty())
            {
                s += '-';
                need--;
            }
            if(need == 0){
                while(!stk.empty())
                {
                    char tmp = stk.top();
                    stk.pop();
                    s += tmp;
                }
            }
        }
        reverse(s.begin(), s.end());
        return s;
    }
};
// @lc code=end

